### PR TITLE
Adds test for LazyEntryViewFromRecord

### PR DESCRIPTION
**Pull request by ahmetmircik**

### CODE REVIEW

**Comment by eminn** [comment]
👍

File: hazelcast/src/main/java/com/hazelcast/map/impl/record/CachedDataRecordWithStats.java [hunk 1]
```diff
@@ -52,4 +52,31 @@ public Object getCachedValueUnsafe() {
...
         return CACHED_VALUE.compareAndSet(this, expectedValue, newValue);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/map/impl/record/CachedDataRecordWithStats.java]: it won't work with subclasses. is it desirable in *Record classes?

Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/record/CachedDataRecordWithStats.java]: I don't have a counter example so it seem ok to me.

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**